File: src/main/java/de/fluxparticle/animation/signal/SignalTransform.java
package de.fluxparticle.animation.signal;

public class SignalTransform<T> extends SignalFunction<T> {

	private final double timeStart;

	private final double timeDuration;

	private final SignalPath<T> path;

	public SignalTransform(double timeStart, double timeDuration, SignalPath<T> path) {
		this.timeStart = timeStart;
		this.timeDuration = timeDuration;
		this.path = path;
	}

	@Override
	public T apply(Double t) {
		return path.apply((t - timeStart) / timeDuration);
	}
}

File: common/src/main/kotlin/de/fluxparticle/animation/signal/SignalTransform.kt
package de.fluxparticle.animation.signal

class SignalTransform<T>(private val timeStart: Double, private val timeDuration: Double, private val path: SignalPath<T>) : SignalFunction<T>() {

    override fun apply(t: Double): T {
        return path.apply((t - timeStart) / timeDuration)
    }

}
